### PR TITLE
Fix `sqrt(epsilon(1._dp))`

### LINKED ISSUE #1122: sqrt(epsilon(1._dp)) does not work
This:
```fortran
program test
integer, parameter :: dp = kind(0.d0)
real(dp) :: tol
tol = sqrt(epsilon(1._dp))
print *, tol
end
```
Gives:
```console
$ gfortran a.f90 && ./a.out 
   1.4901161193847656E-008
$ lfortran a.f90 
1.49166814624004135e-154
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1122.
With this, we get [`minpack/example_lmdif1`](https://github.com/certik/minpack/blob/scipy21/examples/example_lmdif1.f90) aligned with GFortran by every bit.

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I am not sure why "Test WASM" fails, if anyone can have a look.

**Comment by Smit-create** [comment]
From the test failure, I think WASM fails because of `error stop` in the file. Not sure, need to take a closer look though.

File: integration_tests/intrinsics_34.f90 [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+    real(dp) :: y = 2.33
     print *, epsilon(x)
     print *, epsilon(y)
-end program
\ No newline at end of file
+    print *, sqrt(epsilon(1._dp)) !Part of Minpack
+
+    ! Below numbers are corresponding output of gfortran and not magic numbers.
+    if (abs(epsilon(x) - 1.19209290E-07) > 1e-8) error stop
+    if (abs(epsilon(y) - 2.2204460492503131E-016) > 1e-8) error stop
```
Comment (certik) [inline integration_tests/intrinsics_34.f90:11]:
```suggestion
    if (abs(epsilon(y) - 2.2204460492503131E-016) > 1e-15) error stop
```

File: integration_tests/intrinsics_34.f90 [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
-    real(8) :: y = 2.33
+    real(dp) :: y = 2.33
     print *, epsilon(x)
     print *, epsilon(y)
-end program
\ No newline at end of file
+    print *, sqrt(epsilon(1._dp)) !Part of Minpack
+
+    ! Below numbers are corresponding output of gfortran and not magic numbers.
+    if (abs(epsilon(x) - 1.19209290E-07) > 1e-8) error stop
```
Comment (certik) [inline integration_tests/intrinsics_34.f90:10]:
```suggestion
    if (abs(epsilon(x) - 1.19209290E-07) > 1e-7) error stop
```

File: integration_tests/intrinsics_34.f90 [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
+    if (abs(epsilon(y) - 2.2204460492503131E-016) > 1e-8) error stop
+    if (abs(sqrt(epsilon(1._dp)) - 1.4901161193847656E-008) > 1e-8) error stop
```
Comment (certik) [inline integration_tests/intrinsics_34.f90:12]:
```suggestion
    if (abs(sqrt(epsilon(1._dp)) - 1.4901161193847656E-008) > 1e-15) error stop
```

**Comment by certik** [comment]
The WASM failure must be resolved.

**Comment by Pranavchiku** [comment]
It passes with following diff
```diff
diff --git a/integration_tests/intrinsics_34.f90 b/integration_tests/intrinsics_34.f90
index 29d973494..f2ec6305a 100644
--- a/integration_tests/intrinsics_34.f90
+++ b/integration_tests/intrinsics_34.f90
@@ -4,10 +4,10 @@ program intrinsics_34
     real(dp) :: y = 2.33
     print *, epsilon(x)
     print *, epsilon(y)
-    print *, sqrt(epsilon(1._dp)) !Part of Minpack
+    ! print *, sqrt(epsilon(1._dp)) !Part of Minpack
 
     ! Below numbers are corresponding output of gfortran and not magic numbers.
     if (abs(epsilon(x) - 1.19209290E-07) > 1e-7) error stop
     if (abs(epsilon(y) - 2.2204460492503131E-016) > 1e-15) error stop
-    if (abs(sqrt(epsilon(1._dp)) - 1.4901161193847656E-008) > 1e-15) error stop
+    ! if (abs(sqrt(epsilon(1._dp)) - 1.4901161193847656E-008) > 1e-15) error stop
 end program
 ```
If we look at the error it is 
```console
warning: WASM: Calls to C Intrinsic Functions are not yet supported
 --> /home/pranavchiku/lfortran/integration_tests/intrinsics_34.f90:1:1
  |
1 | program intrinsics_34
  | ^ Function: calls c_dp_rand_num

code generation error: RealBinop: only x**2 implemented so far for powers
```
which gives the idea that `sqrt` is not yet supported. Correct me if I am wrong.

**Comment by Pranavchiku** [comment]
Pardon me for messing up this PR, I opened a new PR. This won't happen again.

**Pull request closed without merging**